scapy/fields.py

```diff
@@ -404,136 +488,180 @@ class PadField(object):
     __slots__ = ["_fld", "_align", "_padwith"]
 
     def __init__(self, fld, align, padwith=None):
+        # type: (Field[Any, Any], int, Optional[bytes]) -> None
         self._fld = fld
         self._align = align
         self._padwith = padwith or b"\x00"
 
     def padlen(self, flen):
+        # type: (int) -> int
         return -flen % self._align
 
-    def getfield(self, pkt, s):
+    def getfield(self,
+                 pkt,  # type: BasePacket
+                 s,  # type: bytes
+                 ):
+        # type: (...) -> Tuple[bytes, Any]
         remain, val = self._fld.getfield(pkt, s)
         padlen = self.padlen(len(s) - len(remain))
         return remain[padlen:], val
 
-    def addfield(self, pkt, s, val):
+    def addfield(self,
+                 pkt,  # type: BasePacket
+                 s,  # type: bytes
+                 val,  # type: Any
+                 ):
+        # type: (...) -> bytes
         sval = self._fld.addfield(pkt, b"", val)
-        return s + sval + struct.pack("%is" % (self.padlen(len(sval))), self._padwith)  # noqa: E501
+        return s + sval + struct.pack(
+            "%is" % (
+                self.padlen(len(sval))
+            ),
+            self._padwith
+        )
 
     def __getattr__(self, attr):
+        # type: (str) -> Any
         return getattr(self._fld, attr)
 
 
 class ReversePadField(PadField):
     """Add bytes BEFORE the proxified field so that it starts at the specified
        alignment from its beginning"""
 
-    def getfield(self, pkt, s):
+    def getfield(self,
+                 pkt,  # type: BasePacket
+                 s,  # type: bytes
+                 ):
+        # type: (...) -> Tuple[bytes, Any]
         # We need to get the length that has already been dissected
         padlen = self.padlen(len(pkt.original) - len(s))
         remain, val = self._fld.getfield(pkt, s[padlen:])
         return remain, val
 
-    def addfield(self, pkt, s, val):
+    def addfield(self,
+                 pkt,  # type: BasePacket
+                 s,  # type: bytes
+                 val,  # type: Any
+                 ):
+        # type: (...) -> bytes
         sval = self._fld.addfield(pkt, b"", val)
-        return s + struct.pack("%is" % (self.padlen(len(s))), self._padwith) + sval  # noqa: E501
+        return s + struct.pack("%is" % (
+            self.padlen(len(s))
+        ), self._padwith) + sval
 
 
-class FCSField(Field):
+class FCSField(Field[int, int]):
```

Comment:
Does this work with Python2?

Reply:
Yes thanks to the `__getitem__` trick in `base_classes.py`

Reply:
Cool.